### PR TITLE
[Mobile app] Search feature redesign

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- Search feature redesign

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

#### Screenshots (optional)
hjhj

### CODE REVIEW

File: mobile/lib/models/search_history.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import 'air_quality_reading.dart';
+import 'hive_type_id.dart';
+
+part 'search_history.g.dart';
+
+@JsonSerializable()
+@HiveType(typeId: searchHistoryTypeId)
+class SearchHistory extends HiveObject with EquatableMixin {
+  SearchHistory({
```
Comment (NoahNsimbe) [inline mobile/lib/models/search_history.dart:13]: Search history model

File: mobile/lib/utils/extensions.dart [hunk 2]
```diff
@@ -110,11 +111,49 @@ extension AnalyticsListExt on List<Analytics> {
   }
 }
 
+extension SearchHistoryListExt on List<SearchHistory> {
+  List<SearchHistory> sortByDateTime({bool latestFirst = true}) {
```
Comment (NoahNsimbe) [inline mobile/lib/utils/extensions.dart:115]: Sorting search history

File: mobile/lib/widgets/custom_shimmer.dart [hunk 3]
```diff
@@ -206,110 +206,3 @@ class TextLoadingAnimation extends StatelessWidget {
     );
   }
 }
-
-class AnalyticsCardLoading extends StatelessWidget {
-  const AnalyticsCardLoading({super.key});
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/custom_shimmer.dart:211]: Removing unused widgets

File: mobile/lib/services/location_service.dart [hunk 4]
```diff
@@ -124,9 +124,7 @@ class LocationService {
       }
 
       final sortedReadings =
-          sortAirQualityReadingsByDistance(nearestAirQualityReadings)
-              .take(top)
-              .toList();
+          nearestAirQualityReadings.sortByDistance().take(top).toList();
```
Comment (NoahNsimbe) [inline mobile/lib/services/location_service.dart:127]: Using extensions

**Review by BenjaminSsempala — APPROVED** [review]
Tested this out. Works well on my end

**Review by Mnoble-19 — APPROVED** [review]

File: mobile/lib/services/hive_service.dart [hunk 5]
```diff
@@ -182,11 +203,12 @@ class HiveService {
...
   static String get appNotifications => 'appNotifications';
+  static String get searchHistory => 'searchHistory';
   static String get kya => 'kya';
   static String get profile => 'profile';
   static String get encryptionKey => 'hiveEncryptionKey';
   static String get analytics => 'analytics';
-  static String get airQualityReadings => 'airQualityReadingsBox';
-  static String get nearByAirQualityReadings => 'nearByAirQualityReadings';
+  static String get airQualityReadings => 'airQualityReadings-v1';
+  static String get nearByAirQualityReadings => 'nearByAirQualityReading-v1';
```
Comment (NoahNsimbe) [inline mobile/lib/services/hive_service.dart:212]: Creating new hive storage boxes for air quality readings. Previous boxes got corrupted when we removed hardcoded regions in [this PR.](https://github.com/airqo-platform/AirQo-frontend/pull/979#discussion_r1033732029) hence causing the app to crush on startup. @Baalmart

**Review by Baalmart — APPROVED** [review]
This LGTM, thanks @NoahNsimbe 

I have left a few comments for your review....

File: mobile/lib/screens/map/map_view_widgets.dart [hunk 6]
```diff
@@ -711,7 +711,7 @@ class SearchSites extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final filteredAirQualityReadings =
-        filterNearestLocations(airQualityReadings);
+        airQualityReadings.filterNearestLocations();
```
[the remark this answers is not in the document]
Comment (NoahNsimbe) [reply mobile/lib/screens/map/map_view_widgets.dart:714]: The distance is 4 km @Baalmart

File: mobile/lib/services/location_service.dart [hunk 7]
```diff
@@ -259,6 +257,6 @@ class LocationService {
       }
     }
 
-    return sortAirQualityReadingsByDistance(nearestSites);
+    return nearestSites.sortByDistance();
```
[the remark this answers is not in the document]
Comment (NoahNsimbe) [reply mobile/lib/services/location_service.dart:260]: It's 4 km

**Comment by NoahNsimbe** [comment]
`All the subsequent recent searches were not captured`.  Thanks @Baalmart  shall investigate

File: mobile/lib/services/location_service.dart (hunk 7 above)
[the remark this answers is not in the document]
Comment (NoahNsimbe) [reply mobile/lib/services/location_service.dart:260]: @Baalmart 
https://github.com/airqo-platform/AirQo-frontend/pull/993#discussion_r1054141639

**Pull request merged**